ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AvailableStorageTable.test.tsx

```diff
@@ -273,6 +273,45 @@ describe("AvailableStorageTable", () => {
     expect(wrapper.find("AddLogicalVolume").exists()).toBe(true);
   });
 
+  it("can open the edit logical volume form", () => {
+    const volumeGroup = diskFactory({ type: DiskTypes.VOLUME_GROUP });
+    const logicalVolume = diskFactory({
+      available_size: MIN_PARTITION_SIZE + 1,
+      parent: { id: volumeGroup.id, type: volumeGroup.type, uuid: "vg0" },
+      type: DiskTypes.VIRTUAL,
+    });
+    const state = rootStateFactory({
+      machine: machineStateFactory({
+        items: [
+          machineDetailsFactory({
+            disks: [volumeGroup, logicalVolume],
+            system_id: "abc123",
+          }),
+        ],
+        statuses: machineStatusesFactory({
+          abc123: machineStatusFactory(),
+        }),
+      }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <AvailableStorageTable canEditStorage systemId="abc123" />
+      </Provider>
+    );
+
+    wrapper.find("TableMenu button").at(1).simulate("click");
```

Comment:
It probably safer to target a test class here. Incase we add another button to the pattern.

Reply:
Yep you're right, fixing it up now

Reply:
Actually this requires a change to the `ContextualMenu` react-component first so it allows normal html attributes